cdap-ui/app/cdap/components/SecureKeys/SecureKeyCreate/index.tsx

```diff
@@ -0,0 +1,208 @@
+
+  const [showData, setShowData] = React.useState(false);
+
+  const onLocalNameChange = (e) => {
+    setLocalName(e.target.value);
+  };
+
+  const onLocalDescriptionChange = (e) => {
+    setLocalDescription(e.target.value);
+  };
+
+  const onLocalDataChange = (e) => {
+    setLocalData(e.target.value);
+  };
+
+  const onLocalPropertiesChange = (keyvalue) => {
+    setLocalPropertiesInString(keyvalue);
+  };
+
+  const convertLocalPropertiesInString = (keyvalue: string) => {
+    let keyvaluePairs = Map({});
+    keyvalue.split(COMMON_DELIMITER).forEach((pair) => {
+      const [key, value] = pair.split(COMMON_KV_DELIMITER);
+      keyvaluePairs = keyvaluePairs.set(key, value);
+    });
+    return keyvaluePairs;
+  };
+
+  const saveSecureKey = () => {
+    // Duplicate key name should raise an error
+    const keyIDs = secureKeys.map((key) => key.get('name'));
+    if (keyIDs.includes(localName)) {
+      dispatch({ type: 'SET_SECURE_KEY_STATUS', secureKeyStatus: SecureKeyStatus.Failure });
+      return;
+    }
+
+    addSecureKey();
+  };
+
+  const addSecureKey = () => {
+    const namespace = getCurrentNamespace();
+
+    const params = {
+      namespace,
+      key: localName,
+    };
+
+    const requestBody = {
+      description: localDescription,
+      data: localData,
+      properties: convertLocalPropertiesInString(localPropertiesInString),
+    };
+
+    MySecureKeyApi.put(params, requestBody).subscribe(() => {
+      setLocalName('');
+      setLocalDescription('');
+      setLocalData('');
+      setLocalPropertiesInString('');
+      dispatch({ type: 'SET_SECURE_KEY_STATUS', secureKeyStatus: SecureKeyStatus.Success });
+      handleClose();
+    });
+  };
+
+  return (
+    <Dialog open={open} onClose={handleClose}>
+      <DialogTitle>Create secure key</DialogTitle>
+      <DialogContent>
+        <div className={classnames(classes.margin, classes.textField)}>
+          <TextField
+            variant="outlined"
+            label="Name"
+            defaultValue={localName}
+            onChange={onLocalNameChange}
+            InputProps={{
+              className: classes.textField,
+            }}
+          />
+        </div>
+        <div className={classnames(classes.margin, classes.textField)}>
+          <TextField
+            variant="outlined"
+            label="Description"
+            defaultValue={localDescription}
+            onChange={onLocalDescriptionChange}
+            InputProps={{
+              className: classes.textField,
+            }}
+          />
+        </div>
+        <div className={classnames(classes.margin, classes.textField)}>
+          <TextField
+            variant="outlined"
+            label="Data"
+            type={showData ? 'text' : 'password'}
+            value={localData}
+            onChange={onLocalDataChange}
+            InputProps={{
+              className: classes.textField,
+            }}
+          />
```

Comment:
the textfield for the data should allow multiline

Reply:
I don't think password textfield can be multiline.